pandas/io/parsers.py

```diff
@@ -1421,6 +1422,53 @@ def __init__(self, kwds):
         # keep references to file handles opened by the parser itself
         self.handles = []
 
+    def _validate_parse_dates_presence(self, columns: List[str]) -> None:
+        """
+        Check if parse_dates are in columns.
+
+        If user has provided names for parse_dates, check if those columns
+        are available.
+
+        Parameters
+        ----------
+        columns : list
+            list of names of the dataframe.
+
+        Raises
+        ------
+        ValueError
+            If column to parse_date is not in dataframe.
+
+        """
+        if isinstance(self.parse_dates, list):
+            # a column in parse_dates could be represented
+            # ColReference = Union[int, str]
+            # DateGroups = List[ColReference]
+            # ParseDates = Union[ DateGroups, List[DateGroups],
+            #     Dict[ColReference, DateGroups]]
+            cols_needed = chain.from_iterable(
+                [col if isinstance(col, list) else [col] for col in self.parse_dates]
+            )
+        elif isinstance(self.parse_dates, dict):
+            cols_needed = chain(*self.parse_dates.values())
+        else:
+            cols_needed = chain()
```

Comment:
```suggestion
            cols_needed = []
```

Reply:
If I do this, `mypy` throws error. 

> pandas/io/parsers.py:1455: error: Incompatible types in assignment (expression has type "List[<nothing>]", variable has type "Iterator[Any]")

Reply:
`iter([])` would be my preferred option then, using chain when not chaining anything, while working, feels hacky, and confusing while reading the code.

Reply:
Mypy considers the initial assignment as the definition of a variable. see https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#type-inference

The initial assignment of cols_needed is chain.from_iterable(...) and therefore variable has type "Iterator[Any]"

You can override the inferred type of a variable by using a variable type annotation. see https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#explicit-types-for-variables

i.e. add `cols_needed: Iterable` at L1443 and then can do https://github.com/pandas-dev/pandas/pull/32320#discussion_r390359511 without mypy error.